src/scene/batching.js

```diff
@@ -0,0 +1,719 @@
+pc.extend(pc, function () {
+
+    var Batch = function (meshInstances, isDynamic) {
+        this.origMeshInstances = meshInstances;
+        this._aabb = new pc.BoundingBox();
+        this.meshInstance = null;
+        this.model = null;
+        this.isDynamic = isDynamic;
+    };
+
+    var BatchGroup = function (id, name, isDynamic, maxAabbSize) {
+        this.isDynamic = isDynamic;
+        this.maxAabbSize = maxAabbSize;
+        this.id = id;
+        this.name = name;
+    };
+
+    // Modified SkinInstance for batching
+    // Doesn't contain bind matrices, simplier
+    var SkinBatchInstance = function (device, nodes, rootNode) {
+        this.device = device;
+        this.rootNode = rootNode;
+        this._dirty = true;
+
+        // Unique per clone
+        this.bones = nodes;
+
+        var numBones = nodes.length;
+
+        if (device.supportsBoneTextures) {
+            var size;
+            if (numBones > 256)
+                size = 64;
+            else if (numBones > 64)
+                size = 32;
+            else if (numBones > 16)
+                size = 16;
+            else
+                size = 8;
+
+            this.boneTexture = new pc.Texture(device, {
```

Comment:
Same here, are these "actually" bones?

Reply:
Technically there's no difference between skinning bones and batching nodes, they work exactly the same way.